packages/account/src/BiconomySmartAccount.ts

```diff
@@ -40,7 +40,7 @@ import {
 import { Signer } from 'ethers'
 
 export class BiconomySmartAccount extends SmartAccount implements IBiconomySmartAccount {
-  private factory!: SmartAccountFactory_v100
+  private factory!: any
```

Comment:
I was getting below error

```
@biconomy/account: src/BiconomySmartAccount.ts(166,5): error TS2322: Type 'SmartAccountFactory_v100 | SmartAccountFactory_v200' is not assignable to type 'SmartAccountFactory_v100'.
@biconomy/account:   Type 'SmartAccountFactory_v200' is not assignable to type 'SmartAccountFactory_v100'.
@biconomy/account:     The types of 'interface.functions' are incompatible between these types.
@biconomy/account:       Type '{ "accountCreationCode()": FunctionFragment; "addStake(address,uint32)": FunctionFragment; "basicImplementation()": FunctionFragment; ... 8 more ...; "withdrawStake(address,address)": FunctionFragment; }' is missing the following properties from type '{ "accountCreationCode()": FunctionFragment; "basicImplementation()": FunctionFragment; "deployAccount(address)": FunctionFragment; "deployCounterFactualAccount(address,uint256)": FunctionFragment; "getAddressForCounterFactualAccount(address,uint256)": FunctionFragment; "minimalHandler()": FunctionFragment; }': "deployAccount(address)", "deployCounterFactualAccount(address,uint256)", "getAddressForCounterFactualAccount(address,uint256)"
@biconomy/account: error Command failed with exit code 2.
@biconomy/account: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build exited 2 in '@biconomy/account'
```

because

this.factory = getSAFactoryContract(factoryInstanceDto) 

comes from common and now return type there is changed as below,

```
export function getSAFactoryContract(contractInstanceDto: GetContractInstanceDto): BaseContract {
  const { smartAccountType, version, contractAddress, provider } = contractInstanceDto

  switch (version) {
    case 'V1_0_0':
      if (smartAccountType === SmartAccountType.BICONOMY) {
        return SmartAccountFactory_v100__factory.connect(contractAddress, provider)
      }
      break
    case 'V2_0_0':
      if (smartAccountType === SmartAccountType.BICONOMY) {
        return SmartAccountFactory_v200__factory.connect(contractAddress, provider)
      }
      break
    default:
      return SmartAccountFactory_v100__factory.connect(contractAddress, provider)
  }
  throw new Error('Invalid version or smartAccountType provided for factory contract instance')
}
```

which is causing the compatibility issue

Reply:
I could maybe just return ethers BaseContract and give all contract types as BaseContract.

Reply:
pushing the fix

Reply:
done